Tests/LibJS/test262-runner.cpp

```diff
@@ -0,0 +1,791 @@
+                error.details = message.to_string_without_side_effects();
+        }
+        if (error.type.is_empty())
+            error.type = error_value.to_string_without_side_effects();
+        return error;
+    }
+    return {};
+}
+
+static HashMap<String, String> s_cached_harness_files;
+
+static Result<StringView, TestError> read_harness_file(StringView harness_file)
+{
+    auto cache = s_cached_harness_files.find(harness_file);
+    if (cache == s_cached_harness_files.end()) {
+        auto file = Core::File::construct(String::formatted("{}{}", s_harness_file_directory, harness_file));
+        if (!file->open(Core::OpenMode::ReadOnly)) {
+            return TestError {
+                NegativePhase::Harness,
+                "filesystem",
+                String::formatted("Could not open file: {}", harness_file),
+                harness_file
+            };
+        }
+
+        auto contents = file->read_all();
+        StringView contents_view = contents;
+        s_cached_harness_files.set(harness_file, contents_view.to_string());
+        cache = s_cached_harness_files.find(harness_file);
+        VERIFY(cache != s_cached_harness_files.end());
+    }
+    return cache->value.view();
+}
+
+static Result<JS::NonnullGCPtr<JS::Script>, TestError> parse_harness_files(JS::Realm& realm, StringView harness_file)
+{
+    auto source_or_error = read_harness_file(harness_file);
+    if (source_or_error.is_error())
+        return source_or_error.release_error();
+    auto program_or_error = parse_program<JS::Script>(realm, source_or_error.value(), harness_file);
+    if (program_or_error.is_error()) {
+        return TestError {
+            NegativePhase::Harness,
+            program_or_error.error().type,
+            program_or_error.error().details,
+            harness_file
+        };
+    }
+    return program_or_error.release_value().get<JS::NonnullGCPtr<JS::Script>>();
+}
+
+enum class StrictMode {
+    Both,
+    NoStrict,
+    OnlyStrict
+};
+
+static constexpr auto sta_harness_file = "sta.js"sv;
+static constexpr auto assert_harness_file = "assert.js"sv;
+static constexpr auto async_include = "doneprintHandle.js"sv;
+
+struct TestMetadata {
+    Vector<StringView> harness_files { sta_harness_file, assert_harness_file };
+
+    StrictMode strict_mode { StrictMode::Both };
+    JS::Program::Type program_type { JS::Program::Type::Script };
+    bool is_async { false };
+
+    bool is_negative { false };
+    NegativePhase phase { NegativePhase::ParseOrEarly };
+    StringView type;
+};
+
+static Result<void, TestError> run_test(StringView source, StringView filepath, TestMetadata const& metadata)
+{
+    if (s_parse_only || (metadata.is_negative && metadata.phase == NegativePhase::ParseOrEarly && metadata.program_type != JS::Program::Type::Module)) {
+        // Creating the vm and interpreter is heavy so we just parse directly here.
+        // We can also skip if we now the test is supposed to fail during parse
```

Comment:
```suggestion
        // We can also skip if we know the test is supposed to fail during parse
```

oops, one more

Reply:
I don't now why I keep making this mistake know